framework/src/org/checkerframework/framework/stub/StubParser.java

```diff
@@ -1506,6 +1507,55 @@ private Object getValueOfExpressionInAnnotation(
     }
 
     /**
+     * Returns the TypeElement with the fully qualified name {@code name}, if one exists. Otherwise,
+     * checks {@code enclosingClass} and {@code packageName} for such a class with {@code name}.
+     *
+     * @param packageName name of package being parsed
+     * @param enclosingClass fully-qualified name of the class being parsed
+     * @param name classname (Simple or fully qualified)
+     * @return the TypeElement for {@code name}
+     */
+    private TypeElement findTypeOfName(String packageName, String enclosingClass, String name) {
+        TypeElement typeElement = elements.getTypeElement(name);
+        if (typeElement != null) {
+            importedTypes.put(name, typeElement);
+            return typeElement;
+        }
+
+        while (!enclosingClass.equals(packageName)) {
+            typeElement = elements.getTypeElement(enclosingClass + "." + name);
+            if (typeElement != null) {
+                importedTypes.put(enclosingClass + "." + name, typeElement);
+                return typeElement;
+            }
+
+            int lastDot = enclosingClass.lastIndexOf('.');
+            if (lastDot == -1) {
+                enclosingClass = "";
```

Comment:
When is `""` useful as value for `enclosingClass`? As it will later be appended with `"."`, no legal class name can be produced. Instead, should this break out of the loop?